base/commands/migration/start_stages.go

```diff
@@ -0,0 +1,219 @@
+		if err != nil {
+			return err
+		}
+		st.statusMap, err = st.ci.Client().GetMap(ctx, MakeStatusMapName(st.migrationID))
+		if err != nil {
+			return err
+		}
+		st.updateTopic, err = st.ci.Client().GetTopic(ctx, MakeUpdateTopicName(st.migrationID))
+		if err != nil {
+			return err
+		}
+		st.updateMsgChan = make(chan UpdateMessage)
+		st.topicListenerID, err = st.updateTopic.AddMessageListener(ctx, st.topicListener)
+		return err
+	}
+}
+
+func (st *StartStages) topicListener(event *hazelcast.MessagePublished) {
+	var u UpdateMessage
+	err := json.Unmarshal(event.Value.(serialization.JSON), &u)
+	if err != nil {
+		panic(fmt.Errorf("receiving update from migration cluster: %w", err))
+	}
+	st.updateMsgChan <- u
+}
+
+func (st *StartStages) startStage(ctx context.Context, ec plug.ExecContext) func(stage.Statuser) error {
+	return func(stage.Statuser) error {
+		cb, err := makeConfigBundle(st.configDir, st.migrationID)
+		if err != nil {
+			return err
+		}
+		if err = st.startQueue.Put(ctx, cb); err != nil {
+			return err
+		}
+		ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+		defer cancel()
+		if isTerminal, err := st.handleUpdateMessage(ctx, ec, <-st.updateMsgChan); isTerminal {
+			return err
+		}
+		return nil
+	}
+}
+
+func makeConfigBundle(configDir, migrationID string) (serialization.JSON, error) {
+	var cb ConfigBundle
+	cb.MigrationID = migrationID
+	if err := cb.Walk(configDir); err != nil {
+		return nil, err
+	}
+	b, err := json.Marshal(cb)
+	if err != nil {
+		return nil, err
+	}
+	return b, nil
+}
+
+func (st *StartStages) migrateStage(ctx context.Context, ec plug.ExecContext) func(statuser stage.Statuser) error {
+	return func(stage.Statuser) error {
+		for {
+			select {
+			case msg := <-st.updateMsgChan:
+				if isTerminal, err := st.handleUpdateMessage(ctx, ec, msg); isTerminal {
+					return err
+				}
+			case <-ctx.Done():
+				if err := ctx.Err(); err != nil {
+					if errors.Is(err, context.DeadlineExceeded) {
+						return timeoutErr
+					}
+					return fmt.Errorf("migration failed: %w", err)
+				}
+			}
+		}
+	}
+}
+
+func (st *StartStages) handleUpdateMessage(ctx context.Context, ec plug.ExecContext, msg UpdateMessage) (bool, error) {
+	ec.PrintlnUnnecessary(msg.Message)
+	if slices.Contains([]Status{StatusComplete, StatusFailed, StatusCanceled}, msg.Status) {
+		ms, err := readMigrationStatus(ctx, st.statusMap)
+		if err != nil {
+			return true, fmt.Errorf("reading status: %w", err)
+		}
+		ec.PrintlnUnnecessary(ms.Report)
+		name := fmt.Sprintf("migration_report_%s", st.migrationID)
+		if err = saveReportToFile(name, ms.Report); err != nil {
+			return true, fmt.Errorf("writing report to file: %w", err)
+		}
+		if err = st.saveDebugLogs(ctx, st.ci.OrderedMembers()); err != nil {
+			return true, fmt.Errorf("writing debug logs to file: %w", err)
+		}
+		ec.PrintlnUnnecessary(fmt.Sprintf("migration report saved to file: %s", name))
+		for _, l := range ms.Logs {
+			ec.Logger().Info(l)
+		}
```

Comment:
We should not log all logs at the end of migration IMO. We already print them in real time.

Reply:
I think @kutluhanmetin wanted to make the logs appear also in the CLC log, which is the correct approach.
But that won't be necessary if we use the CLC log exclusively during the migration.